File: client-ios/SerenadaCore/Sources/Models/SerenadaPeerConnectionState.swift
import Foundation

/// Typed peer connection state, replacing raw String.
/// Values match WebRTC RTCPeerConnectionState (UPPER_CASE for cross-platform wire parity with Android).
public enum SerenadaPeerConnectionState: String, Codable, Equatable, Sendable {
    case new = "NEW"
    case connecting = "CONNECTING"
    case connected = "CONNECTED"
    case disconnected = "DISCONNECTED"
    case failed = "FAILED"
    case closed = "CLOSED"
}

File: client-android/serenada-core/src/main/java/app/serenada/core/call/SerenadaPeerConnectionState.kt
package app.serenada.core.call

import org.webrtc.PeerConnection

/** Per-peer connection state exposed on [RemoteParticipant]. */
enum class SerenadaPeerConnectionState(val value: String) {
    NEW("NEW"),
    CONNECTING("CONNECTING"),
    CONNECTED("CONNECTED"),
    DISCONNECTED("DISCONNECTED"),
    FAILED("FAILED"),
    CLOSED("CLOSED");

    companion object {
        fun fromRtcState(state: PeerConnection.PeerConnectionState): SerenadaPeerConnectionState =
            when (state) {
                PeerConnection.PeerConnectionState.NEW -> NEW
                PeerConnection.PeerConnectionState.CONNECTING -> CONNECTING
                PeerConnection.PeerConnectionState.CONNECTED -> CONNECTED
                PeerConnection.PeerConnectionState.DISCONNECTED -> DISCONNECTED
                PeerConnection.PeerConnectionState.FAILED -> FAILED
                PeerConnection.PeerConnectionState.CLOSED -> CLOSED
            }
    }
}
